### PR TITLE
DOCSP-15810: update changestream page to use new feature + cleanup

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - [DOCSP-30151](https://jira.mongodb.org/browse/DOCSP-30151)
[Staging](https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/DOCSP-15810-update-cs-example/fundamentals/crud/read-operations/change-streams/)

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: source/fundamentals/crud/read-operations/change-streams.txt [hunk 1]
```diff
@@ -125,15 +124,18 @@ update operations:
    :start-after: begin aggregationExample
    :end-before: end aggregationExample
 
-When the change stream receives an update change event, the preceding code
-example outputs the following text:
+An update operation on the collection should produce output similar to the
+following text:
```
Comment (mongoKart) [inline source/fundamentals/crud/read-operations/change-streams.txt:128]: I: [Style Guide on "should"](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#s:~:text=the%20setup%20procedure.-,should,-Use%20should%20only)

S: The text you deleted looks fine to me. Otherwise:

```suggestion
An update operation on the collection produces the following output:
```

File: source/fundamentals/crud/read-operations/change-streams.txt [hunk 2]
```diff
@@ -142,22 +144,23 @@ Include Pre-images and Post-images
 
 You can configure the change event to contain or omit the following data:
 
-- The **pre-image** which is a document that represents the version of the
-  document before the operation if it exists
-- The **post-image** which is a document that represents the version of the
-  document after the operation if it exists
+- The **pre-image**, which is a document that represents the version of the
+  document before the operation, if it exists
+- The **post-image**, which is a document that represents the version of the
```
Comment (mongoKart) [inline source/fundamentals/crud/read-operations/change-streams.txt:149]: S: Sounds lighter and cleaner to omit "which is"

```suggestion
- The **pre-image**, a document that represents the version of the
  document before the operation, if it exists
- The **post-image**, a document that represents the version of the
```

File: source/fundamentals/crud/read-operations/change-streams.txt [hunk 3]
```diff
@@ -142,22 +144,23 @@ Include Pre-images and Post-images
...
+  document after the operation, if it exists
 
 To receive change stream events that include a pre-image or post-image, you
-must connect to a MongoDB v6.0 or later deployment and set up the following:
+must connect to a deployment running MongoDB v6.0 or later and perform the
+following actions:
```
Comment (mongoKart) [inline source/fundamentals/crud/read-operations/change-streams.txt:154]: S: This sentence is a little hard to parse. Could we move the version info to an admonition or something?

```suggestion
must perform perform the following actions on your MongoDB deployment:
```

File: source/fundamentals/crud/read-operations/change-streams.txt [hunk 4]
```diff
@@ -168,19 +171,19 @@ must connect to a MongoDB v6.0 or later deployment and set up the following:
...
+     events, see the :ref:`<java-post-image-example>`.
 
 .. _java-change-stream-pre-post-collection:
 
 Create a Collection with Pre-Image and Post-Images Enabled
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-To create a collection with the pre-image and post-image option using the
-driver, specify an instance of  ``ChangeStreamPreAndPostImagesOptions``
+To use the driver to create a collection that has the pre-image and post-image options
```
Comment (mongoKart) [inline source/fundamentals/crud/read-operations/change-streams.txt:185]: S: 
```suggestion
To use the driver to create a collection with the pre-image and post-image options
```

File: source/fundamentals/crud/read-operations/change-streams.txt [hunk 5]
```diff
@@ -191,22 +194,23 @@ and call the ``createCollection()`` method as shown in the following example:
...
+perform this operation, see the entry on :manual:`collMod </reference/command/collMod/>`
+in the Server manual.
 
 .. warning::
 
    When you modify this option on a collection, any change streams open on
-   that collection in your application may fail if configured to require
-   receiving the pre-image or post-image.
+   that collection in your application may fail if you configure them to require
+   the pre-image or post-image.
```
Comment (mongoKart) [inline source/fundamentals/crud/read-operations/change-streams.txt:204]: S: Might be able to trim this down a bit. [Style Guide](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#s:~:text=than%20matrices.-,may,-Use%20may%20to) also recommends using "may" only for permission

Maybe something like: 
"If you enable this option while change streams are open in your application, 
those change streams might fail."

File: source/fundamentals/crud/read-operations/change-streams.txt [hunk 6]
```diff
@@ -215,23 +219,27 @@ include the pre-image and output the results:
...
+``FullDocumentBeforeChange.REQUIRED`` option. This option configures the change
+stream to require pre-images for replace, update, and delete change
+events. If the pre-image is not available, the driver raises an error.
 
-Suppose an application updated the ``latestVersion`` field of a document in a
-collection of software library dependencies from the value of ``2.0.0`` to
-``2.1.0``. The corresponding change event output by the preceding code example
-should resemble the following text:
+Suppose you update the ``amount`` field of a document from ``150`` to
+``2000``. The change event produced by this change should appear in the
```
Comment (mongoKart) [inline source/fundamentals/crud/read-operations/change-streams.txt:227]:
```suggestion
``2000``. The change event produced by this change appears in the
```

File: source/fundamentals/crud/read-operations/change-streams.txt [hunk 7]
```diff
@@ -251,25 +260,27 @@ include the post-image and output the results:
    :end-before: end fullDocumentExample
 
 The preceding example configures the change stream to use the
-``FullDocument.UPDATE_LOOKUP`` option. This configures the change
-stream to return both the deltas between the original and changed document
-and a copy of the document at some point in time after the change occurred.
+``FullDocument.WHEN_AVAILABLE`` option. This option configures the change
+stream to the change stream to return the post-image of the modified
+document for replace and update change events, if it is available.
```
Comment (mongoKart) [inline source/fundamentals/crud/read-operations/change-streams.txt:265]:
```suggestion
``FullDocument.WHEN_AVAILABLE`` option. This option configures the change
stream to return the post-image of the modified
document for replace and update change events, if it's available.
```

File: source/fundamentals/crud/read-operations/change-streams.txt [hunk 8]
```diff
@@ -251,25 +260,27 @@ include the post-image and output the results:
...
 
-Suppose an application updated the ``population`` field of a document from
-the value of ``800`` to ``950`` in a collection of city census data. The
-corresponding change event output by the preceding code example should
-resemble the following text:
+Suppose you update the ``color`` field of a document from ``"purple"`` to
```
Comment (mongoKart) [inline source/fundamentals/crud/read-operations/change-streams.txt:267]: S: Could say "value of the color field," but this reads fine to me.

**Review by mongoKart — CHANGES_REQUESTED** [review]
few small things! looks good otherwise.

File: source/fundamentals/crud/read-operations/change-streams.txt (hunk 1 above)
Comment (rustagir) [reply source/fundamentals/crud/read-operations/change-streams.txt:128]: changed everywhere

File: source/fundamentals/crud/read-operations/change-streams.txt (hunk 5 above)
Comment (rustagir) [reply source/fundamentals/crud/read-operations/change-streams.txt:204]: reworded

File: source/fundamentals/crud/read-operations/change-streams.txt (hunk 6 above)
Comment (rustagir) [reply source/fundamentals/crud/read-operations/change-streams.txt:227]: changed along with changes from an earlier comment

**Review by mongoKart — APPROVED** [review]
lgtm
